### PR TITLE
Clean ML benchmark

### PR DESCRIPTION
**Pull request by fteytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

I think this implementation is more in a "sklearn style". It may be easier to other people to use it.

## How Has This Been Tested (if it applies)

I have tested it only with a small optimization. It could be great to test it by launching experiments seqmltuning and naiveseqmltuning.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [ ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/ml/mlfunctionlib.py [hunk 1]
```diff
@@ -189,18 +195,20 @@ def make_dataset(self, data_dimension: tp.Optional[int], dataset: str) -> None:
...
-                valid_range = np.arange(self.num_data) % 10 == cv
-                self.X_train += [self.X[train_range]]
-                self.y_train += [self.y[train_range]]
-                self.X_valid += [self.X[valid_range]]
-                self.y_valid += [self.y[valid_range]]
-            self.X_test = data[0][1::2]
-            self.y_test = data[1][1::2]
+            self.X = data[0]
+            self.y = data[1]
+            self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y, test_size=0.5, random_state=42)
```
Comment (jrapin) [inline nevergrad/functions/ml/mlfunctionlib.py:200]: `random_state=42`: do we want it to be deterministic? using the randomstate linked to the parametrization would make the function stochastic while keeping the reproducibility property

**Review by jrapin — APPROVED** [review]
My personal feeling is that there are way too many variables here, so it is probably lacking some structure to make things easier (eg using a Dataset named tuple for pairing X, y together, or use a booleans to index X and y to recover X_train and X_test etc.

But that's more work for later, nevermind this for now. The only possible "issue/feature" i see would be the use of a deterministic/seeded split. It depends how you want the function to behave.

**Comment by jrapin** [comment]
Also you'll need to update the values in the tests ;)

**Comment by teytaud** [comment]
The error for "install" is super weird to me... maybe we just rerun...

**Comment by fteytaud** [comment]
Ok I have changed the test values, sorry I always forget that.

For the number of variables, I agree but I can't find an easy way to simplify this, as there are many ml experiments in the class.

File: nevergrad/functions/ml/mlfunctionlib.py (hunk 1 above)
Comment (fteytaud) [reply nevergrad/functions/ml/mlfunctionlib.py:200]: It was deterministic, so i let it.
If we change that, we have to change the tests.

**Comment by fteytaud** [comment]
> The error for "install" is super weird to me... maybe we just rerun...

I don't know :(
I have updated the tests, they are ok now, but the install error is still there.

**Comment by jrapin** [comment]
> > The error for "install" is super weird to me... maybe we just rerun...
> 
> I don't know :(
> I have updated the tests, they are ok now, but the install error is still there.

I'll investigate, looks unrelated

**Comment by jrapin** [comment]
@fteytaud can you try merging master? There were some incompatibilities with recent sphinx (doc building) versions

**Comment by fteytaud** [comment]
> @fteytaud can you try merging master? There were some incompatibilities with recent sphinx (doc building) versions

Seems ok, you're a boss ;-)

**Comment by jrapin** [comment]
merged, thanks ;)

**Pull request merged**